crates/data-transformer/tests/bytes31.rs

```diff
@@ -0,0 +1,78 @@
+use data_transformer::cairo_types::{CairoBytes31, ParseBytes31Error};
+use starknet_types_core::felt::Felt;
+use std::str::FromStr;
+
+#[cfg(test)]
+mod tests_bytes31 {
+    use super::*;
+
+    #[test]
+    fn test_happy_case() {
+        let bytes31 = CairoBytes31::from_str("0x123456789abcdef").unwrap();
+        assert_eq!(
+            Felt::from(bytes31),
+            Felt::from_hex_unchecked("0x123456789abcdef")
+        );
+    }
+
+    #[test]
+    fn test_max_value() {
+        let max_bytes31 = CairoBytes31::MAX;
+        let from_str = CairoBytes31::from_str(
+            "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+        )
+        .unwrap();
+        assert_eq!(max_bytes31, from_str);
+    }
+
+    #[test]
+    fn test_overflow() {
+        let result = CairoBytes31::from_str(
+            "0x1ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+        );
+        assert!(matches!(result, Err(ParseBytes31Error::Overflow)));
+    }
+
+    #[test]
+    fn test_invalid_hex() {
+        let result = CairoBytes31::from_str("invalid_hex");
+        assert!(matches!(result, Err(ParseBytes31Error::CairoFromStrError)));
+    }
+
+    #[test]
+    fn test_empty_string() {
+        let result = CairoBytes31::from_str("");
+        assert!(matches!(result, Err(ParseBytes31Error::CairoFromStrError)));
+    }
```

Comment:
```suggestion
    #[test_case("wrong_string_value"; "wrong string value")]
    #[test_case(""; "empty string")]
    fn test_invalid_string(input: &str) {
        let result = CairoBytes31::from_str(input);
        assert!(matches!(result, Err(ParseBytes31Error::CairoFromStrError)));
    }
```

Also, we need to add `test-case` under `[dev-dependencies]` in `Cargo.toml`.